src/controller/CommissioneeDeviceProxy.cpp

```diff
@@ -0,0 +1,333 @@
+                   ChipLogDetail(Controller, "Connection expired, but it doesn't match the current session"));
+    mState = ConnectionState::NotConnected;
+    mSecureSession.ClearValue();
+}
+
+CHIP_ERROR CommissioneeDeviceProxy::OnMessageReceived(Messaging::ExchangeContext * exchange, const PayloadHeader & payloadHeader,
+                                                      System::PacketBufferHandle && msgBuf)
+{
+    if (mState == ConnectionState::SecureConnected)
+    {
+        if (mStatusDelegate != nullptr)
+        {
+            mStatusDelegate->OnMessage(std::move(msgBuf));
+        }
+        else
+        {
+            HandleDataModelMessage(exchange, std::move(msgBuf));
+        }
+    }
+    return CHIP_NO_ERROR;
+}
+
+void CommissioneeDeviceProxy::OnResponseTimeout(Messaging::ExchangeContext * ec) {}
+
+CHIP_ERROR CommissioneeDeviceProxy::CloseSession()
+{
+    ReturnErrorCodeIf(mState != ConnectionState::SecureConnected, CHIP_ERROR_INCORRECT_STATE);
+    if (mSecureSession.HasValue())
+    {
+        mSessionManager->ExpirePairing(mSecureSession.Value());
+    }
+    mState = ConnectionState::NotConnected;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CommissioneeDeviceProxy::UpdateAddress(const Transport::PeerAddress & addr)
+{
+    bool didLoad;
+
+    mDeviceAddress = addr;
+
+    ReturnErrorOnFailure(LoadSecureSessionParametersIfNeeded(didLoad));
+
+    if (!mSecureSession.HasValue())
+    {
+        // Nothing needs to be done here.  It's not an error to not have a
+        // secureSession.  For one thing, we could have gotten an different
+        // UpdateAddress already and that caused connections to be torn down and
+        // whatnot.
+        return CHIP_NO_ERROR;
+    }
+
+    Transport::SecureSession * secureSession = mSessionManager->GetSecureSession(mSecureSession.Value());
+    secureSession->SetPeerAddress(addr);
+
+    return CHIP_NO_ERROR;
+}
+
+void CommissioneeDeviceProxy::Reset()
+{
+    SetActive(false);
+
+    mState          = ConnectionState::NotConnected;
+    mSessionManager = nullptr;
+    mStatusDelegate = nullptr;
+    mInetLayer      = nullptr;
+#if CONFIG_NETWORK_LAYER_BLE
+    mBleLayer = nullptr;
+#endif
+    if (mExchangeMgr)
+    {
+        // Ensure that any exchange contexts we have open get closed now,
+        // because we don't want them to call back in to us after this
+        // point.
+        mExchangeMgr->CloseAllContextsForDelegate(this);
+    }
+    mExchangeMgr = nullptr;
+
+    ReleaseDAC();
+    ReleasePAI();
+}
+
+CHIP_ERROR CommissioneeDeviceProxy::LoadSecureSessionParameters()
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    PASESession pairingSession;
+
+    if (mSessionManager == nullptr || mState == ConnectionState::SecureConnected)
+    {
+        ExitNow(err = CHIP_ERROR_INCORRECT_STATE);
+    }
+
+    if (mState == ConnectionState::Connecting)
+    {
+        ExitNow(err = CHIP_NO_ERROR);
+    }
+
+    err = pairingSession.FromSerializable(mPairing);
```

Comment:
We should no longer have a need to ever serialize / store / load / deserialize mPairing.  Can we remove all of the code that does this?

Reply:
Thanks!